core/src/tests/recent_job_results_list.cc

```diff
@@ -32,7 +32,40 @@
 #include <iostream>
 #include <fstream>
 
-static auto Is32BitWordsize = []() { return sizeof(void*) == 4; };
+static inline bool Is32BitWordsize()
+{
+#if defined HAVE_WIN32 and defined HAVE_MINGW
+  return false;  // wincross always 64Bit
+#else
+  return sizeof(void*) == 4;
+#endif
+};
+
+TEST(recent_job_results_list, check_header_padding_bytes)
+{
+  std::string detected;
+
+  if (Is32BitWordsize()) {
+    detected = "32Bit wordsize";
+    EXPECT_EQ(offsetof(StateFileHeader, version), 16);
+    EXPECT_EQ(offsetof(StateFileHeader, last_jobs_addr), 20);
+    EXPECT_EQ(offsetof(StateFileHeader, end_of_recent_job_results_list), 28);
+    EXPECT_EQ(offsetof(StateFileHeader, reserved), 36);
+  } else {
+    detected = "64Bit wordsize";
+    EXPECT_EQ(offsetof(StateFileHeader, version), 16);
+    EXPECT_EQ(offsetof(StateFileHeader, last_jobs_addr), 24);
+    EXPECT_EQ(offsetof(StateFileHeader, end_of_recent_job_results_list), 32);
+    EXPECT_EQ(offsetof(StateFileHeader, reserved), 40);
+  }
+
+#if HAVE_WIN32
+  detected += ", Windows";
+#endif
+  detected += " detected";
+
+  std::cout << " <<< " << detected << " >>> " << std::endl;
+}
```

Comment:
In my opinion, this doesn't make sense. Most of this can (and should) be checked at compile-time. No need for a runtime check.
The premise of `Is32BitWordsize()` is just wrong. The default struct packing is implementation specific. While it may rely on the width of the processor registers, this is nothing you can rely on.

The different location of `last_jobs_addr` on Windows has nothing to do with pointer width or anything. It is plainly based on the different [default alignment for Windows](https://docs.microsoft.com/en-us/cpp/c-language/storage-and-alignment-of-structures?view=vs-2019).
On most 64-bit systems the int32_t will be aligned as 64-bits, on most 32-bit systems it will be aligned as 32-bits.
For Windows the rule is different and completely ignores the architecture: Each member of a struct is aligned as the largest member of the struct. So a struct containing only int16_t will be aligned as 16-bits, even on a 64-bit machine.
In this case, alignment will be 64 bits on windows, no matter what.

```suggestion
static inline bool Is32BitAligned()
{
  return offsetof(StateFileHeader, last_jobs_addr) == 20;
};
```